nova_vm/src/ecmascript/types/language/bigint.rs

```diff
@@ -278,6 +387,53 @@ impl BigInt {
         }
     }
 
+    /// ### [6.1.6.2.6 BigInt::remainder ( n, d )](https://tc39.es/ecma262/#sec-numeric-types-bigint-remainder)
+    pub(crate) fn remainder(agent: &mut Agent, n: BigInt, d: BigInt) -> JsResult<BigInt> {
+        match (n, d) {
+            (BigInt::SmallBigInt(n), BigInt::SmallBigInt(d)) => {
+                if d == SmallBigInt::zero() {
+                    return Err(agent.throw_exception_with_static_message(
+                        ExceptionType::RangeError,
+                        "Division by zero",
+                    ));
+                }
+                let (n, d) = (n.into_i64() as i128, d.into_i64() as i128);
+                let result = n % d;
+
+                if let Ok(result) = SmallInteger::try_from(result) {
+                    Ok(BigInt::SmallBigInt(SmallBigInt(result)))
+                } else {
+                    Ok(agent.heap.create(BigIntHeapData {
+                        data: result.into(),
+                    }))
+                }
+            }
+            (BigInt::SmallBigInt(n), BigInt::BigInt(d)) => {
+                let n = n.into_i64();
+                let d = &agent[d];
+                Ok(agent.heap.create(BigIntHeapData { data: n % &d.data }))
```

Comment:
**issue:** This result, as well as the `(BigInt, BigInt)` one, could fit in the `SmallBigInt` range.